src/devices/StUsb4500/Objects/BatteryObject.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using UnitsNet;
+using UnitsNet.Units;
+
+namespace Iot.Device.StUsb4500.Objects
+{
+    /// <summary>
+    /// PDO of a battery sources.
+    /// </summary>
+    public class BatteryObject : PowerDeliveryObject
+    {
+        private const uint OperatingPowerMask = 0b0011_1111_1111;
+        private const uint MinVoltageMask = 0b1111_1111_1100_0000_0000;
+        private const uint MaxVoltageMask = 0b0011_1111_1111_0000_0000_0000_0000_0000;
+
+        /// <summary>Gets or sets the operating power.</summary>
+        /// <remarks>This is stored with the factor 4 as a 10-bit value (range 0 - 1023) => 0...255.75W.</remarks>
+        public Power OperatingPower
+        {
+            get => Power.FromWatts((ushort)(Value & OperatingPowerMask) / 4.0);
+            set
+            {
+                CheckArgumentInRange(value.Value, 255.75M);
```

Comment:
Thanks for adding the test for the arguments. You should use, and that's for everywhere value.TheUnitToTest (replace TheUnitToTest by the real one). value.Value will return the value in which it has been created. So imagine you create in millivolts, you'll et values which are 1000 times higher and it will be rejected.
So same for the next lines, you'll need to get them with the target unit first.
This apply everywhere unfortunately.
See here for Power for example: https://github.com/angularsen/UnitsNet/blob/a7e55169673f68ef25f15e3f0305f23dc894a6f4/UnitsNet/GeneratedCode/Quantities/Power.g.cs#L162